File: app/src/main/java/com/karyaprestasi/fishpal/ui/main_ui/profile/ProfileFragment.java
//package com.project.fishbud.ui.main_ui.profile;
//
//import android.os.Bundle;
//import android.view.LayoutInflater;
//import android.view.View;
//import com.project.fishbud.R;
//import android.view.ViewGroup;
//
//import androidx.annotation.Nullable;
//import androidx.fragment.app.Fragment;
//import androidx.viewpager.widget.ViewPager;
//
//import com.google.android.material.tabs.TabLayout;
//import com.project.fishbud.ui.viewPager.AkunPembeliFragment;
//import com.project.fishbud.ui.viewPager.AkunTokoFragment;
//import com.project.fishbud.ui.viewPager.SectionPagerAdapter;
//
//import java.lang.reflect.Field;
//
//public class ProfileFragment extends Fragment {
//
//    View myFragment;
//
//    ViewPager viewPager;
//    TabLayout tabLayout;
//
//
//    public ProfileFragment() {
//        // Required empty public constructor
//    }
//
//    public static ProfileFragment getInstance()    {
//        return new ProfileFragment();
//    }
//
//
//    @Override
//    public View onCreateView(LayoutInflater inflater, ViewGroup container,
//                             Bundle savedInstanceState) {
//        // Inflate the layout for this fragment
//        myFragment = inflater.inflate(R.layout.fragment_profile2, container, false);
//
//        viewPager = myFragment.findViewById(R.id.viewPager);
//        tabLayout = myFragment.findViewById(R.id.tabLayout);
//
//        return myFragment;
//    }
//
//    //Call onActivity Create method
//
//
//    @Override
//    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
//        super.onActivityCreated(savedInstanceState);
//
//        setUpViewPager(viewPager);
//        tabLayout.setupWithViewPager(viewPager);
//
//        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
//            @Override
//            public void onTabSelected(TabLayout.Tab tab) {
//
//            }
//
//            @Override
//            public void onTabUnselected(TabLayout.Tab tab) {
//
//            }
//
//            @Override
//            public void onTabReselected(TabLayout.Tab tab) {
//
//            }
//        });
//    }
//
//    private void setUpViewPager(ViewPager viewPager) {
//        SectionPagerAdapter adapter = new SectionPagerAdapter(getChildFragmentManager());
//
//        adapter.addFragment(new AkunPembeliFragment(), "Akun Pembeli");
//        adapter.addFragment(new AkunTokoFragment(), "Akun Toko");
//
//        viewPager.setAdapter(adapter);
//    }
//
////    @Override
////    public void onDetach() {
////        super.onDetach();
////        try {
////            Field childFragmentManager = Fragment.class
////                    .getDeclaredField("mChildFragmentManager");
////            childFragmentManager.setAccessible(true);
////            childFragmentManager.set(this, null);
////        } catch (NoSuchFieldException e) {
////            throw new RuntimeException(e);
////        } catch (IllegalAccessException e) {
////            throw new RuntimeException(e);
////        }
////    }
//}

File: app/src/main/java/com/karyaprestasi/fishpal/ui/main_ui/profile/ProfileFragment.kt
package com.karyaprestasi.fishpal.ui.main_ui.profile

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.google.android.material.tabs.TabLayoutMediator
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.karyaprestasi.fishpal.databinding.BelumLoginBinding
import com.karyaprestasi.fishpal.databinding.FragmentProfileBinding
import com.karyaprestasi.fishpal.ui.authentication.AuthenticationActivity
import com.karyaprestasi.fishpal.ui.viewPager.SectionsPagerAdapter

class ProfileFragment : Fragment() {

    private lateinit var binding: FragmentProfileBinding
    private lateinit var bindingLogin: BelumLoginBinding
    private lateinit var adapter: SectionsPagerAdapter
    private var user: FirebaseUser? = null

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        user = FirebaseAuth.getInstance().currentUser
        if (user == null) {
            bindingLogin = BelumLoginBinding.inflate(layoutInflater, container, false)
            return bindingLogin.root
        } else {
            binding = FragmentProfileBinding.inflate(layoutInflater, container, false)
            return binding.root
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        adapter = SectionsPagerAdapter(childFragmentManager, lifecycle)

        if (user != null) {
            binding.viewPager2.adapter = adapter
            binding.viewPager2.isSaveEnabled = false

            TabLayoutMediator(binding.tabs, binding.viewPager2) { tab, position ->
                when (position) {
                    0 -> {
                        tab.text = "Akun Pembeli"
                    }
                    1 -> {
                        tab.text = "Akun Toko"
                    }
                }
            }.attach()
        } else {
            bindingLogin.btnLoginProfile.setOnClickListener {
                val move = Intent(context, AuthenticationActivity::class.java)
                move.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
                context?.startActivity(Intent(context, AuthenticationActivity::class.java))
            }
        }
    }

//    override fun onBackPressed(): Boolean {
//        return false
//    }
}